<launch>
  <include file="$(find openni_launch)/launch/openni.launch" />
  <include file="$(find jsk_pcl_ros)/launch/pointcloud_screenpoint.launch">
    <arg name="USE_VIEW" value="true" />
    <arg name="PUBLISH_POINTS" value="true" />
    <arg name="image" value="/camera/rgb"/>
    <arg name="points" value="/camera/depth_registered/points"/>
    <arg name="image_type" default="image_color"/>
  </include>

  <node pkg="nodelet" type="nodelet" name="voxel_grid"
        args="load pcl/VoxelGrid /screenpoint_manager"
        output="screen"
        clear_params="true"
        respawn="false">
    <remap from="~input" to="/camera/depth_registered/points"/>
    <rosparam>
      filter_limit_min: 0.1
      filter_limit_max: -0.1
      filter_limit_negative: true
      leaf_size: 0.01
    </rosparam>p
  </node>
  
  <node pkg="nodelet" type="nodelet" name="ground_segmentation" args="load pcl/SACSegmentation /screenpoint_manager"
        output="screen"
        clear_params="true"
        respawn="false">
        <remap from="~input" to="voxel_grid/output" />
        <rosparam>
          # -[ Mandatory parameters
          model_type: 9
          distance_threshold: 0.01
          max_iterations: 5000
          method_type: 0
#          axis: [0.0, -1.0, 0.0]
          optimize_coefficients: true
          max_queue_size: 20
          min_inliers: 100
        </rosparam>
  </node>

  
  
  <node pkg="nodelet" type="nodelet" name="extract_nonground_indices" args="load pcl/ExtractIndices /screenpoint_manager" output="screen" respawn="true">
    <remap from="~input"   to="voxel_grid/output" />
    <remap from="~indices" to="/ground_segmentation/inliers" />
    <rosparam>
      negative: true
      max_queue_size: 100
    </rosparam>
  </node>

  
  <node pkg="nodelet"
        type="nodelet"
        args="load jsk_pcl/HintedPlaneDetector /screenpoint_manager"
        name="hinted_plane_detector"
        clear_params="true">
    <remap from="~input" to="/extract_nonground_indices/output" />
    <remap from="~hint" to="/pointcloud_screenpoint_nodelet/output" />
  </node>
  <node pkg="jsk_pcl_ros"
        type="marker_appender.py"
        name="marker_appender">
    <remap from="marker" to="/hinted_plane_detector/marker" />
  </node>
</launch>
